woocommerce/woocommerce-ios
# Orders tab (iOS 12 & iOS 13+): add site ID predicate for Orders

Precaution for #2830 

Before this PR, there is no predicate on site ID in Orders tab FRC. After #2830, we're not tearing down the persistent store and there could be remaining data that are saved after logging out. Having site ID as part of the FRC predicate is a good way to make sure any leftover data from another site do not get used for the currently logged in site. I did a pass of FRCs in the code base and just found the Orders tab, plus two models `SiteVisitStats` and `TopEarnerStats` don't have any site ID info. This PR focuses on the FRC for the Orders tab (2 implementations).

This change also allows us to support keeping orders for multiple sites in the future.

## Changes

- In https://github.com/woocommerce/woocommerce-ios/commit/b599ff9ed0cd43f1fbb5238bae9d9381a8ad6f97 (iOS 12 implementation, soon to be deprecated): refactored how a predicate is created in `OrdersViewModel`, and created a function `refreshOrdersPredicate` that is called when site changes in `OrdersViewController`'s observation. Note that after https://github.com/woocommerce/woocommerce-ios/issues/1838, this `refreshOrdersPredicate` won't be necessary anymore. From my testing, `.defaultAccountWasUpdated` does not get triggered after switching sites and `.StoresManagerDidUpdateDefaultSite` does.
- In https://github.com/woocommerce/woocommerce-ios/commit/5e92137089f8d700f20e460df77ce42d704e50a1: similar changes for iOS 13+ implementation.

## Testing

No user-facing changes are expected, please test for confidence check:

- Go to the Orders tab --> make sure both tabs show the same orders as before
- Go to My store tab and > Settings, and log out from the app
- Go to the Orders tab --> make sure both tabs show the orders from the new site
- Log out from the app
- Log in to a store
- Go to the Orders tab --> make sure both tabs show the orders from the new site

Update release notes:
